src/shims/x86/gfni.rs

```diff
@@ -0,0 +1,215 @@
+            // See `affine_transform` for details.
+            // https://www.intel.com/content/www/us/en/docs/intrinsics-guide/index.html#text=gf2p8affine_
+            "vgf2p8affineqb.128" | "vgf2p8affineqb.256" | "vgf2p8affineqb.512" => {
+                let [left, right, imm8] =
+                    this.check_shim(abi, Abi::C { unwind: false }, link_name, args)?;
+                affine_transform(this, left, right, imm8, dest, false)?;
+            }
+            // Used to implement the `_mm{, 256, 512}_gf2p8affineinv_epi64_epi8` functions.
+            // See `affine_transform` for details.
+            // https://www.intel.com/content/www/us/en/docs/intrinsics-guide/index.html#text=gf2p8affineinv
+            "vgf2p8affineinvqb.128" | "vgf2p8affineinvqb.256" | "vgf2p8affineinvqb.512" => {
+                let [left, right, imm8] =
+                    this.check_shim(abi, Abi::C { unwind: false }, link_name, args)?;
+                affine_transform(this, left, right, imm8, dest, true)?;
+            }
+            // Used to implement the `_mm{, 256, 512}_gf2p8mul_epi8` functions.
+            // Multiplies packed 8-bit integers in `left` and `right` in the finite field GF(2^8)
+            // and store the results in `dst`. The field GF(2^8) is represented in
+            // polynomial representation with the reduction polynomial x^8 + x^4 + x^3 + x + 1.
+            // https://www.intel.com/content/www/us/en/docs/intrinsics-guide/index.html#text=gf2p8mul
+            "vgf2p8mulb.128" | "vgf2p8mulb.256" | "vgf2p8mulb.512" => {
+                let [left, right] =
+                    this.check_shim(abi, Abi::C { unwind: false }, link_name, args)?;
+
+                let (left, left_len) = this.project_to_simd(left)?;
+                let (right, right_len) = this.project_to_simd(right)?;
+                let (dest, dest_len) = this.project_to_simd(dest)?;
+
+                assert_eq!(left_len, right_len);
+                assert_eq!(dest_len, right_len);
+
+                for i in 0..dest_len {
+                    let left =
+                        u32::from(this.read_scalar(&this.project_index(&left, i)?)?.to_u8()?);
+                    let right =
+                        u32::from(this.read_scalar(&this.project_index(&right, i)?)?.to_u8()?);
+                    let dest = this.project_index(&dest, i)?;
+
+                    // Step 1: Carryless multiplication of `left` and `right`.
+                    // Carryless multiplication is explained in the `pclmulqdq`function.
+                    let mut result = 0u32;
+                    for i in 0..8 {
+                        if left & (1 << i) != 0 {
+                            result ^= right << i;
+                        }
+                    }
+
+                    // Step 2: Polynomial division modulo 2.
+                    // Polynomial division modulo 2 is defined inside the `sse42` module.
```

Comment:
It is? Where?

Reply:
CRC definition.

Reply:
If I Ctrl-F "CRC definition" in the `sse42` module, there is no result. So I don't know what you mean by that.

Please fix this comment to be a reference that unambiguously references the correct part of `sse42` in a stable way.

Reply:
The intrinsic names are now mentioned explicitly.